controllers/bindingtarget/objectmarker.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/commaseparated"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type BindingTargetObjectMarker struct {
+}
+
+// LinkAnnotation is used to associate the binding to the service account (even the referenced service accounts get annotated by this so that we can clean up their secret lists when the binding is deleted).
+const LinkAnnotation = "spi.appstudio.redhat.com/linked-access-token-binding" //#nosec G101 -- false positive, this is just a label
+
+// ManagedByBindingLabel marks the other objects as managed by SPI. Meaning that their lifecycle is bound
+// to the lifecycle of some SPI binding.
+const ManagedByBindingLabel = "spi.appstudio.redhat.com/managed-by-binding"
+
+var _ bindings.ObjectMarker = (*BindingTargetObjectMarker)(nil)
+
+// IsManaged implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) IsManagedBy(ctx context.Context, binding client.ObjectKey, obj client.Object) (bool, error) {
+	refed, _ := m.IsReferencedBy(ctx, binding, obj)
+	return refed && obj.GetLabels()[ManagedByBindingLabel] == binding.Name, nil
+}
+
+// IsReferenced implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) IsReferencedBy(ctx context.Context, binding client.ObjectKey, obj client.Object) (bool, error) {
+	annos := obj.GetAnnotations()
+	if len(annos) == 0 {
+		return false, nil
+	}
+	anno := annos[LinkAnnotation]
+	return commaseparated.Value(anno).Contains(binding.Name), nil
+}
+
+// ListManagedOptions implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) ListManagedOptions(ctx context.Context, binding client.ObjectKey) ([]client.ListOption, error) {
+	return []client.ListOption{client.MatchingLabels{ManagedByBindingLabel: binding.Name}}, nil
+}
+
+// ListReferencedOptions implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) ListReferencedOptions(ctx context.Context, binding client.ObjectKey) ([]client.ListOption, error) {
+	// the link is represented using an annotation, so we cannot produce any list options
+	return []client.ListOption{}, nil
+}
+
+// MarkManaged implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) MarkManaged(ctx context.Context, binding client.ObjectKey, obj client.Object) (bool, error) {
+	changed, _ := m.MarkReferenced(ctx, binding, obj)
+
+	labels := obj.GetLabels()
+
+	val := labels[ManagedByBindingLabel]
+
+	if val != binding.Name {
+		changed = true
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		labels[ManagedByBindingLabel] = binding.Name
+		obj.SetLabels(labels)
+	}
+
+	return changed, nil
+}
+
+// MarkReferenced implements dependents.ObjectMarker
+func (m *BindingTargetObjectMarker) MarkReferenced(ctx context.Context, binding client.ObjectKey, obj client.Object) (bool, error) {
+	changed := false
+	annos := obj.GetAnnotations()
+
+	val := commaseparated.Value(annos[LinkAnnotation])
```

Comment:
this has been the case for the last couple of months. A quick google search didn't reveal a max size of annotation value to me but i'm not too worried since for example kubectl stores the last applied value of the whole object in its own annotation.